### DISCUSSION TITLE
Question about Foundry DSC Course

### DISCUSSION THREAD

**Discussion by 0xcontagiante** [Cyfrin/foundry-full-course-cu#17] [Q&A] [answered]
Hello ser Patrick thanks for the amazing courses you are a legend . I have a question to the community regarding how the 1$ peg is actually working like i cant see no code where we set the value of the token to be 1$ can someone please explain to me how we achieve it if i am mistaken please feel free to send me where exactly on the video course  we coded it.

**Comment by cromewar** [comment]
Hello @0xcontagiante thanks for reaching out with your question, please can you provide more context to this, Which lesson are you referring to? can you please add the video time stamp, we are going to be happy to give you support.

**Reply by 0xcontagiante** [reply]
Hello @cromewar thanks for the answer man here is the video:  https://www.youtube.com/watch?v=8dRAd-Bzc_E
To be more precise this phrase exactly: https://github.com/Cyfrin/foundry-defi-stablecoin-f23/blob/main/src/DSCEngine.sol#L36
How can we achieve that our token will maintain 1$ peg without any LPs

**Comment by PatrickAlphaC** [comment, marked as answer]
Great question!!

First off have you watched the content leading up to the stablecoin part? That will give you a lot of context. Secondly here is how it works. 

# Mangos Scenario

Imagine you have 1,000 mangos worth $1,000. (1 mango == $1) However, you need to spend $100 on rent this month. What can you do?

### Easy: Sell 100 mangos for $100 -> then pay rent. 

This makes sense and is a basic approach... but what if you think the price of mangos is about to explode? For example you think the world is about to be on a mango shortage. So you don't want to sell your mangos. Is there a solution you can do? 

There sure is.

## Collateralization 

So you scour the web looking for a solution, and you find one, use your mangos as collateral. 

You find Fred is willing to give you an interesting deal:

Fred: "I'll lend you $100 if you give me $100 of mangos as collateral. Once you pay me my $100 back, I'll give you your mangos back. If you don't pay me back, I will take your 100 mangos. " 

At the time, 100 mangos == $100, so you decide you'll do the deal! You'll give Fred 100 mangos, and he will lend you $100. You then go use this to pay your rent. 

You now have:
- 900 mangos on hand (worth $900)
- -$100 paid for rent
- -$100 in debt to Fred
- 100 mangos as collateral so long as you eventually pay back Fred

## Mangos Explode in price

A few weeks go by, and sure enough, the mango market explodes, and the price of mangos doubles. 1 mango now == $2.

You now have:
- 900 mangos on hand (worth $1800)
- -$100 paid for rent
- -$100 in debt to Fred
- 100 mangos as collateral so long as you eventually pay back Fred (now worth $200) 
Total: $1,900

So you decide that you want to pay back Fred now, and you sell 50 mangos for $100 (since the price is now 2 to 1!) and use that $100 to get your mangos back from Fred.

You now have:
- 1,000 mangos (worth $2,000!)
- -$100 paid for rent
- 0 debt to Fred
- 0 mangos as collateral
Total: $1,900

Wow! Well what if you just sold instead of collateralizing your mangos with Fred?

You'd have:
- 900 mangos (worth $1,800)
- All rent paid
- 0 debt to Fred
- 0 mangos as collateral
Total: $1,800

Wow!! you made out with $100 more doing the deal with Fred! 

# What is this analogous to?

- Fred was the Smart Contract that we deployed. It's a set of immutable logic to promise to give back our mangos if we repay our debt
- The price of mangos -> dollars was based on a Chainlink price feed
- The reason the loan maintained it's price was because we used the mangos to collateralize their value. 

## What was the stablecoin?

Now here is where it can get a little confusing. Let's say instead of Fred giving us $100 he gave us 100 "mango receipts". These receipts are worth $100 because at any time, someone can exchange them for $100 in mangos! The reason the "mango receipts" have value is because they are collateralized by your mangos.

So our stablecoin has value, because we used a price feed to set the rate at which the mango receipts could be exchanged for mangos. 

Let me know if that doesn't make sense...

**Reply by 0xcontagiante** [reply]
Make absolute sense thank you so much for this amazing explanation. I totally understand now how our token is getting 1$ peg without any LPs

**Reply by cromewar** [reply]
Best explanation ever!!

**Reply by alymurtazamemon** [reply]
@PatrickAlphaC Ser! I think the Total would be $1800 and we made out with $100 more doing the deal with Fred!

**Reply by PatrickAlphaC** [reply]
ah! Fixed!

**Reply by Ynyesto** [reply]
I just spent like an hour thinking about this topic, couldn't figure it out and found the answer to my own question while writing it here. I would just delete it, but that would just be time wasted, so here's the post, in case anyone else finds this confusing as well:

Hi, guys! I know it's been a while, but I don't quite get why using a price feed to keep the collateral price updated keeps the peg to $1. Doesn't the value of a stable coin usually stay pegged to USD through arbitrage? I would have thought in order to keep the value of 1 DSC pegged to $1, you'd have to allow anyone with DSC bought in the market redeem it for the same dollarized value in ETH or BTC. That way arbitrage can be done: 
- DSC price goes up to $1.1: arbitragers could, for example, buy 100$ worth of ETH and mint 100 DSC worth 110$. Profit, 10$. Enough arbitrage brings the price of DSC back down to 1$. OK, our system allows this.
- DSC price goes down to $0.9: arbitragers buy, same example, 100 DSC, worth 90 $, burn it and get 100$ worth of ETH in return. Profit, 10$. Enough arbitrage brings the price of DSC back up to 1$. PROBLEM: our system only lets redeeming collateral to whoever has deposited previously, so buying DSC in a DEX and then burning it to get ETH can't be done, right? I mean, of course anyone with DSC can liquidate someone else, but what if there are no users that can be liquidated?

My point is if people swap DSC for USDC en masse, the price of DSC will start to go down (e. g. 1 DSC = 0,9 USDC). Then people who have minted DSC can still burn it to get their ETH, so I suppose to them it's still worth 1 $, but people who BOUGHT their DSC, can't redeem it for ETH... But of course I suppose whoever sold it to them can redeem their collateral, so they would buy back at a lower price the DSC they sold, since they can get the collateral back and still have some DSC left, so that demand would bring the price back up... Hmm, so I suppose the system actually DOES work, and there is no problem with it at all! 😅😅

**Reply by CoraCote** [reply]
This makes absolute sense. Thank you! I get it now: our token can appear “$1-pegged” even without any LPs because the test/price path isn’t actually using real market liquidity.
